test/integration/Subdomains.test.ts

```diff
@@ -0,0 +1,166 @@
+import type { Server } from 'http';
+import fetch from 'cross-fetch';
+import type { Initializer } from '../../src/init/Initializer';
+import type { HttpServerFactory } from '../../src/server/HttpServerFactory';
+import type { ResourceStore } from '../../src/storage/ResourceStore';
+import { joinFilePath } from '../../src/util/PathUtil';
+import { getTestFolder, instantiateFromConfig, removeFolder } from './Config';
+
+const port = 6005;
+const baseUrl = `http://localhost:${port}/`;
+
+const rootFilePath = getTestFolder('subdomains');
+const stores: [string, any][] = [
+  [ 'in-memory storage', {
+    storeUrn: 'urn:solid-server:default:MemoryResourceStore',
+    teardown: jest.fn(),
+  }],
+  [ 'on-disk storage', {
+    storeUrn: 'urn:solid-server:default:FileResourceStore',
+    teardown: (): void => removeFolder(rootFilePath),
+  }],
+];
+
+// Localhost doesn't support subdomains so accessing the subdomain
```

Comment:
It does; `a.localhost` etc. We might be able to fake DNS with a package like nock.
But this solution works too.